crates/provider/src/provider/trait.rs

```diff
@@ -51,6 +51,14 @@ pub trait Provider<T: Transport + Clone = BoxTransport, N: Network = Ethereum>:
     /// Returns the root provider.
     fn root(&self) -> &RootProvider<T, N>;
 
+    /// Returns the [`ProviderBuilder`](crate::ProviderBuilder) to build on.
+    fn builder() -> ProviderBuilder<Identity, Identity, N>
+    where
+        Self: Sized,
+    {
+        ProviderBuilder::default()
+    }
```

Comment:
That would create a cyclic dep

Reply:
adding a Default bound to the Network::Transaction requested assoc type would allow you to write the impl Ethereum block above

Reply:
> adding a Default bound to the Network::TransactionRequest assoc type would allow you to write the impl Ethereum block above

Like so?

```rust
/// The JSON body of a transaction request.
    #[doc(alias = "TxRequest")]
    type TransactionRequest: RpcObject
        + TransactionBuilder<Self>
        + Debug
        + Default
        + From<Self::TxEnvelope>
        + From<Self::UnsignedTx>;
```

How would this avoid a cyclic dep, I'm probably making a noob mistake here.